File: android/src/main/kotlin/com/andrehaueisen/facebook_audience/FacebookConstants.kt
package com.andrehaueisen.facebook_audience

internal object FacebookConstants {
    val MAIN_CHANNEL = "fb.audience.network.io"
    val BANNER_AD_CHANNEL = "$MAIN_CHANNEL/bannerAd"
    val INTERSTITIAL_AD_CHANNEL = "$MAIN_CHANNEL/interstitialAd"
    val NATIVE_AD_CHANNEL = "$MAIN_CHANNEL/nativeAd"
    val REWARDED_VIDEO_CHANNEL = "$MAIN_CHANNEL/rewardedAd"
    val IN_STREAM_VIDEO_CHANNEL = "$MAIN_CHANNEL/inStreamAd"

    val INIT_METHOD = "init"
    val SHOW_INTERSTITIAL_METHOD = "showInterstitialAd"
    val LOAD_INTERSTITIAL_METHOD = "loadInterstitialAd"
    val DESTROY_INTERSTITIAL_METHOD = "destroyInterstitialAd"

    val SHOW_REWARDED_VIDEO_METHOD = "showRewardedAd"
    val LOAD_REWARDED_VIDEO_METHOD = "loadRewardedAd"
    val DESTROY_REWARDED_VIDEO_METHOD = "destroyRewardedAd"

    val DISPLAYED_METHOD = "displayed"
    val DISMISSED_METHOD = "dismissed"
    val ERROR_METHOD = "error"
    val LOADED_METHOD = "loaded"
    val CLICKED_METHOD = "clicked"
    val LOGGING_IMPRESSION_METHOD = "logging_impression"
    val REWARDED_VIDEO_COMPLETE_METHOD = "rewarded_complete"
    val REWARDED_VIDEO_CLOSED_METHOD = "rewarded_closed"

    val IN_STREAM_VIDEO_COMPLETE_METHOD = "in_stream_complete"

    val MEDIA_DOWNLOADED_METHOD = "media_downloaded"
}

File: ios/Classes/FacebookConstants.swift
//
//  FacebookConstants.swift
//  facebook_audience
//
//  Created by Andre Haueisen on 14/06/19.
//

import Foundation

class FacebookConstants {
    static let MAIN_CHANNEL:String = "facebook_audience"
    static let BANNER_AD_CHANNEL:String = "\(MAIN_CHANNEL)/bannerAd"
    static let INTERSTITIAL_AD_CHANNEL:String = "\(MAIN_CHANNEL)/interstitialAd"
    static let NATIVE_AD_CHANNEL:String = "\(MAIN_CHANNEL)/nativeAd"
    static let REWARDED_VIDEO_CHANNEL:String = "\(MAIN_CHANNEL)/rewardedAd"
    static let IN_STREAM_VIDEO_CHANNEL:String = "\(MAIN_CHANNEL)/inStreamAd"
    
    static let INIT_METHOD:String = "init"
    static let SHOW_INTERSTITIAL_METHOD:String = "showInterstitialAd"
    static let LOAD_INTERSTITIAL_METHOD:String = "loadInterstitialAd"
    static let DESTROY_INTERSTITIAL_METHOD:String = "destroyInterstitialAd"
    
    static let SHOW_REWARDED_VIDEO_METHOD:String = "showRewardedAd"
    static let LOAD_REWARDED_VIDEO_METHOD:String = "loadRewardedAd"
    static let DESTROY_REWARDED_VIDEO_METHOD:String = "destroyRewardedAd"
    
    static let DISPLAYED_METHOD:String = "displayed"
    static let DISMISSED_METHOD:String = "dismissed"
    static let ERROR_METHOD:String = "error"
    static let LOADED_METHOD:String = "loaded"
    static let CLICKED_METHOD:String = "clicked"
    static let LOGGING_IMPRESSION_METHOD:String = "logging_impression"
    static let REWARDED_VIDEO_COMPLETE_METHOD:String = "rewarded_complete"
    static let REWARDED_VIDEO_CLOSED_METHOD:String = "rewarded_closed"
    
    static let IN_STREAM_VIDEO_COMPLETE_METHOD:String = "in_stream_complete"
    
    static let MEDIA_DOWNLOADED_METHOD:String = "media_downloaded"
}
